### ISSUE TITLE
Twig bugs

### ISSUE THREAD

**Issue by ausi** [contao/contao#3200]
I noticed some minor bugs in the twig implementation:

~~`{% extends '@Contao/ce_text' %}` does not work `("Nested blocks are not allowed in child templates") in "@Contao_ContaoCoreBundle/ce_text.html5"`. I think this is probably related to `ce_text` itself extending the `block_searchable` template.~~ fixed in #3203

~~`debug:contao-twig` does not show templates from theme folders.~~ fixed in #3219

_The following are not really bugs and might be future scope:_

~~The follwing usages of Contao html5 templates do not work (and I don’t know if the can or should ☺️)~~ see #3213

```
{% extends ['@Contao/…', '@Contao/…'] %}
{% extends true ? '@Contao/…' : '@Contao/…' %}
{% use '@Contao/…' %}
```

~~We could add support for the `html_attr` escaper in the same way as we did for `html` by disabling double_encode.~~ done in #3204

/cc @m-vo

**Comment by m-vo** [comment]
> {% extends '@Contao/ce_text' %} does not work ("Nested blocks are not allowed in child templates") in "@Contao_ContaoCoreBundle/ce_text.html5". I think this is probably related to ce_text itself extending the block_searchable template.

I noticed this as well. This is IMO a limitation of the Contao template engine, but let's see if we can improve this!

**Comment by m-vo** [comment]
> debug:contao-twig does not show templates from theme folders.

This is actually expected. The theme templates aren't part of the template hierarchy (not tracked) because they can only be the first templates in the chain and we would not know which one to pick in case there are multiple themes (runtime dependent). That's also the reason why you cannot include/embed a theme template with `@Contao/my_template`. 

Maybe we can find a solution for both issues/limitations. :thinking:

**Comment by ausi** [comment]
Maybe `debug:contao-twig` could get an argument to specify the theme folder like `--theme=my-theme/with-subfolder` that could make it more obvious to users of this command that themes are handled separately I think.

**Comment by m-vo** [comment]
With #3203, #3204, #3213 and #3219 now the only thing missing from this ticket is support for `{% use '@Contao/…' %}`.

But as the Twig documentation states…

> Horizontal reuse is an advanced Twig feature that is hardly ever needed in regular templates. It is mainly used by projects that need to make template blocks reusable without using inheritance.

… I think it's fine to implement this at a later point in time.

**Comment by ausi** [comment]
Closed in favor of #3203, #3204, #3213 and #3219
